src/transformers/trainer.py

```diff
@@ -137,17 +140,21 @@ class Trainer:
     optimized for 🤗 Transformers.
 
     Args:
-        model (:class:`~transformers.PreTrainedModel`):
-            The model to train, evaluate or use for predictions.
-        args (:class:`~transformers.TrainingArguments`):
-            The arguments to tweak training.
+        model (:class:`~transformers.PreTrainedModel`, `optional`):
+            The model to train, evaluate or use for predictions. If not provided, a ``model_init`` must be passed.
+        args (:class:`~transformers.TrainingArguments`, `optional`):
+            The arguments to tweak training. Will default to a basic instance of :class:`~transformers.TrainingArguments`
+            with the ``output_dir`` set to a directory named `tmp_trainer` in the current directory if not provided.
         data_collator (:obj:`DataCollator`, `optional`, defaults to :func:`~transformers.default_data_collator`):
             The function to use to from a batch from a list of elements of :obj:`train_dataset` or
             :obj:`eval_dataset`.
         train_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
             The dataset to use for training.
         eval_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
             The dataset to use for evaluation.
+        model_init (:obj:`Callable[[], PreTrainedModel]`, `optional`):
```

Comment:
Adding this argument is necessary for hyperparameter search since we need to reinitialize the model before each new run.

Reply:
Should we instead clone the `model` in each hp run? (~same memory/CPU hit than initializing a new one, I guess)

Reply:
You'd need to keep the copy for the next run, so that would take twice the space in memory.

Reply:
Is the RAM (i.e. CPU RAM) footprint of the model usually a bottleneck in trainings? The trade off here is that you complexify the public API a little bit, which might or might not be warranted.

Reply:
Some models take several GB and if the dataset is big, this is definitely going to bother the user. Another solution could be to save the initial model in a tempdir.

Note that I thought the API addition is valuable in itself IMO: I often want to compare some runs in a notebook and having two successive calls to train start from the initial model avoids reinstantiating it. But if it's just me, I can remove.

Reply:
Ok, that sounds reasonable.

Reply:
Note that scanning the `seed` is important in Bert-like-model finetuning and means that we will actually want to have models with different random weights initialization.

Reply:
Thankou so much! But I have a problem when defining the Trainer. It said, "__init__() got an unexpected keyword argument 'model_init'", what should I do? Is the Trainer doesn't detect 'model_init' parameter? Very sorry for the very newbie question :pray:t2: and Thankyou before.